assignments/02-cli-constructive.md

```diff
@@ -3,77 +3,82 @@
-daily.2021-09-24_0010             weekly.2021-09-12_0015
-every_four_hours.2021-09-23_0405  weekly.2021-09-19_0015
-every_four_hours.2021-09-23_0805
-$ 
+
+If only it were that simple for your assignment. We'll get more into all this
+later in the term, but this program is not ideal: it builds `foo.c` into `foo`
+every time `build.sh` is executed, even if `foo.c` has not changed. Your
+program must instead only rebuild `foo` if any of its dependencies have
+changed. In addition, your program must build intermediate object files for
+each C file.
+
+This means example runs might look like this:
+
+```console?prompt=$
+$ ls
+build.sh  foo.c  foo.h  rng.c  rng.h
+$ ./build.sh 
+cc -c foo.c
+cc -c rng.c
+cc -o foo foo.o rng.o
+$ ./build.sh 
+$ touch rng.c
+$ ./build.sh 
+cc -c rng.c
+cc -o foo foo.o rng.o
+$
 ```
 
-This is not a standard feature of Linux or POSIX---it's specific to the network
-servers that Tufts uses. Nevertheless, `.snapshot/` can be very useful when you
-accidentally remove or overwrite an important file and want to get the old
-version back. (In the next module, we'll learn about version control systems,
-which are like `.snapshot/` but supercharged.) In this part of the assignment,
-you'll write a shell script that automates the process of finding what files
-have been added to a directory since its most recent snapshot.
-
-Your script, named `whats-new.sh`, should take as its only argument a path to a
-directory to find new files in. If the argument is missing, or does not point
-to a directory that has been snapshotted, or the script is provided too many
-arguments, your script should print an error message and exit with a nonzero
-exit status. (The `exit` command immediately exits a shell script and takes the
-exit status as an argument.)
-
-If the argument is valid, your script should compare the contents of the
-directory with the contents of the most recent snapshot. We define the *most
-recent snapshot* as the one with the highest timestamp in its name, regardless
-of whether it's `daily`, `weekly`, or something else. We define the *contents*
-of a directory as all the non-hidden files (see note above) that are directly
-contained within it.
-
-In other words, you do not need to show new hidden files (ones that start with
-`.`), nor do you need to recurse into subdirectories. You may do either of
-these if you so choose, however. (If you do decide to handle hidden files, try
-to not list `.snapshot/` itself as a new file!)
+Notice two things:
+
+* We can run `build.sh` and if nothing needs to get rebuilt, nothing will be
+  rebuilt
+* We can update individual C files without recompiling other C files
+
+...
+
+### Requirements
+
+> Note: The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+> "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this
+> document are to be interpreted as described in [RFC
+> 2119](https://datatracker.ietf.org/doc/html/rfc2119).
```

Comment:
Needed? I'm pretty sure students know what "must" means when it occurs in an assignment, and that's the only one you use.